src/shared_modules/content_manager/src/components/gzipDecompressor.hpp

```diff
@@ -0,0 +1,113 @@
+/*
+ * Wazuh content manager
+ * Copyright (C) 2015, Wazuh Inc.
+ * October 20, 2023.
+ *
+ * This program is free software; you can redistribute it
+ * and/or modify it under the terms of the GNU General Public
+ * License (version 2) as published by the FSF - Free Software
+ * Foundation.
+ */
+
+#ifndef _GZIP_DECOMPRESSOR_HPP
+#define _GZIP_DECOMPRESSOR_HPP
+
+#include "json.hpp"
+#include "updaterContext.hpp"
+#include "utils/chainOfResponsability.hpp"
+#include "utils/stringHelper.h"
+#include "utils/zlibHelper.hpp"
+#include <filesystem>
+#include <iostream>
+#include <memory>
+#include <string>
+#include <utility>
+
+/**
+ * @class GzipDecompressor
+ *
+ * @brief Class in charge of decompressing the content in GZ format as a step of a chain of responsibility.
+ *
+ */
+class GzipDecompressor final : public AbstractHandler<std::shared_ptr<UpdaterContext>>
+{
+private:
+    /**
+     * @brief Pushes the state of the current stage into the data field of the context.
+     *
+     * @param contextData Reference to the context data.
+     * @param status Status to be pushed.
+     */
+    void pushStageStatus(nlohmann::json& contextData, std::string status) const
+    {
+        auto statusObject = R"(
+            { 
+                "stage": "GzipDecompressor",
+                "status": "n/a"
+            }
+        )"_json;
+        statusObject.at("status") = std::move(status);
```

Comment:
Improve (performance): Take into account that generating a nlohmann::json object from a literal string has poor performance ([benchmark](https://github.com/wazuh/intelligence-platform/wiki/Take-into-account#nlohmann-performance-boost---object-generation))

```suggestion
        auto statusObject = nlohmann::json::object();
        statusObject["stage"] = "GzipDecompressor";
        statusObject["status"]  = std::move(status);
```

Reply:
Done!